src/Mvc/Mvc.Core/src/ApiExplorer/IApiResponseMetadataProvider.cs

```diff
@@ -11,21 +12,21 @@ namespace Microsoft.AspNetCore.Mvc.ApiExplorer
     /// Provides a return type, status code and a set of possible content types returned by a
     /// successful execution of the action.
     /// </summary>
-    public interface IApiResponseMetadataProvider : IFilterMetadata
+    public interface IApiResponseMetadataProvider : IFilterMetadata, IProducesResponseTypeMetadata
```

Comment:
I don't think we should change the interface, just the conrete types. 

cc @pranavkm any thoughts?

Reply:
> just the concrete types.

👍🏽 makes sense. We can keep this as legacy interface until we can phase it out in favor of the new types.

Reply:
Good call! I've updated the code so that:

* `IApiResponseMetadataProvider` is unchanged
* The `EndpointMetadataApiDescriptionProvider` supports processing both `IApiResponseMetadataProvider` (for attributes on delegates) and the `IProducesResponseTypeMetadata` (for the extension methods).